astropy/units/quantity.py

```diff
@@ -1451,25 +1465,42 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
             },
         }
         formats["latex_inline"] = formats["latex"]
-
-        if format not in formats:
-            raise ValueError(f"Unknown format '{format}'")
-        elif format is None:
-            if precision is None:
+        default_formatter = None
+
+        if format_spec is not None:
+            # Check for format_spec prohibited operators
+            if "%" == format_spec[-1]:
+                raise ValueError("% format_spec is not supported for Quantity.")
+            # Set default formatter
+            default_formatter = {"all": lambda x: f"{x:{format_spec}}"}
+
+        if format is None:
+            # format_spec overwrites precision
+            if default_formatter is None and precision is None:
                 # Use default formatting settings
                 return f"{self.value}{self._unitstr:s}"
-            else:
-                # np.array2string properly formats arrays as well as scalars
-                return (
-                    np.array2string(self.value, precision=precision, floatmode="fixed")
-                    + self._unitstr
+
+            return (
```

Comment:
If we change the `if` as suggested above, then here you could do the definition of `default_formatter` (which is not used elsewhere).